### ISSUE TITLE
Calculate RT breaks when inputs result in large rt value

### ISSUE THREAD

**Issue by rememberlenny** [Recidiviz/covid19-dashboard#407]
**Describe the bug**
The `/calculate_rt` endpoint returns opaque errors when the case data results in large rt values.

This can be very confusing when someone is looking to test the RT value results.

**To Reproduce**

1. Create a new facility and enter `1000 current cases`, and `10000 total population`. (ie. 5/10/20 with 1000 cases)/
2. Go back to the dashboard and add a previous days data. Values should be 1 day before the current date with `100 current cases`. Don't change total population. Save. (ie. 5/9/20 with 100 cases)
3. Add another previous days data, one week before the last day, with `717 cases`. (ie. 5/2/20 with 717 cases)/
4. There should be no issues. Output from calculate_rt endpoint will be the following:

```
`{“Rt”: [{“date”: “2020-05-03”, “value”: 0.0}, {“date”: “2020-05-09", “value”: 12.0}, {“date”: “2020-05-10”, “value”:
8.63}], “high90”: [{“date”: “2020-05-03”, “value”: 10.81}, {“date”: “2020-05-09", “value”: 12.0}, {“date”: “2020-05-10”,
“value”: 9.3}], “low90”: [{“date”: “2020-05-03”, “value”: 0.0}, {“date”: “2020-05-09", “value”: 11.8}, {“date”:
“2020-05-10”, “value”: 7.930000000000001}]}`
```

Now do the same thing as above, but instead the last day of data should be 718 cases. This will result in error.

1. Create a new facility and enter `1000 current cases`, and `10000 total population`. (ie. 5/10/20 with 1000 cases)/
2. Go back to the dashboard and add a previous days data. Values should be 1 day before the current date with `100 current cases`. Don't change total population. Save. (ie. 5/9/20 with 100 cases)
3. Add another previous days data, one week before the last day, with `718 cases`. (ie. 5/2/20 with 718 cases)/
4. This will result in the following error:

```
{"error": "Traceback (most recent call last):\n File \"/user_code/realtime_rt.py\", line 134, in compute_r_t\n hdis =
highest_density_interval(posteriors, p=.9)\n File \"/user_code/realtime_rt.py\", line 89, in highest_density_interval\n
return pd.DataFrame([highest_density_interval(pmf[col], p=p) for col in pmf],\n File \"/user_code/realtime_rt.py\", line
89, in <listcomp>\n return pd.DataFrame([highest_density_interval(pmf[col], p=p) for col in pmf],\n File
	\"/user_code/realtime_rt.py\", line 101, in highest_density_interval\n best = (highs - lows).argmin()\nValueError:
	attempt to get argmin of an empty sequence\n\nDuring handling of the above exception, another exception
	occurred:\n\nTraceback (most recent call last):\n File \"/user_code/helpers.py\", line 44, in wrapped\n
	function_output = f(request_json)\n File \"/user_code/main.py\", line 20, in calculate_rt\n result_df =
	compute_r_t(request_json)\n File \"/user_code/realtime_rt.py\", line 136, in compute_r_t\n raise ValueError('Unable
	to compute R(t) with the provided data')\nValueError: Unable to compute R(t) with the provided data\n"}
```

**Expected behavior**
The python endpoint should give a descriptive error.

Also the front end should make an validation check before approving the case data that results in a broken RT result. This can be done by either checking the caclulate_rt endpoint before submitting the case data.

**Comment by jessex** [comment]
@macfarlandian, it would be good to triage this some time in the next day or two to see what the root cause is at least, but I wouldn't call investigating and fixing this higher priority than what you're finishing now or the scenario sharing work.

**Comment by macfarlandian** [comment]
yeah i spent a good bit of time talking this over with @rememberlenny before this issue was filed ... it's definitely bad but at least it shouldn't affect people in the wild with valid data. Part of what triggers it is having your case counts go down over time, which I think is something we don't expect to have happen? But the bad news is that we are not properly validating that and returning inconsistent results when it happens.

**Comment by macfarlandian** [comment]
@justkunz might have some useful perspective on what's happening inside the model when it gets data like this

**Comment by macfarlandian** [comment]
OK well I have confirmed that the input data must be cumulative case counts (never decreasing over time) so there is a two- or three-pronged approach to resolving this: 

1. validate this on the backend (ensure cases never decrease) and return a more informative error message to that effect (this is easy)
2. validate the form on the frontend by comparing against historical data to keep bad data out of the database (this is considerably more work since there is basically no baseline validation functionality yet AFAIK)
3. (maybe, depends on product input @sychang ) change UI treatment to reduce user error and clarify requirements. flag any already existing invalid data in the historical data view (add cases modal) to let people know which records they need to fix. (imperfect because we can't always infer this with 100% confidence)

I think we can spin these out into separate tickets and prioritize them individually. Personally I would start with backend validation; that plus the error messaging work currently in progress in #280 would at least let you see what went wrong
